### PR TITLE
flash_writer_tool: add support for eMMC erase commands

### PR DESCRIPTION
**Pull request by commodo**

### CODE REVIEW

**Comment by seebe** [comment]
That's an interesting idea.
Let me try it out. If it works, I'll merge it.

Thanks.

**Comment by commodo** [comment]
> That's an interesting idea. Let me try it out. If it works, I'll merge it.
> 
> Thanks.

so, these commands ended up being useful when needing to re-provision some RZG2L boards (during development);
AFAIR, rewriting the boot-partitions did not work too well, without erasing them first;

**Comment by seebe** [comment]
So, good/bad.

Good = The script does just what I expected it to do.
Bad = It didn't erase anything. But, this does not seem to be the script's fault because even when I manually enter the commands, nothing gets erased. So have to figure that out.

But, I would like to ask for these changes:
1: Keep "Show switches" as the last menu item. Put your new commands after "eMMC boot setup"

2: Rename your new menu items to "eMMC erase Boot 1", "eMMC erase Boot 2", "eMMC erase User"

3: Change $OP1 to $OP4 because it's an eMMC command

4: Get rid of the extra space in the menu "x1. $OP1"  ->>  "x1.$OP4" (at some point, we'll get rid of those menu characters altogether)

5: Add your new commands to print_usage()


(or, I can make the changes for you if you really want)

File: flash_writer_tool/flash_writer_tool.sh [hunk 1]
```diff
@@ -1108,6 +1108,9 @@ if [ "$FW_GUI_MODE" == "1" ] ; then
 	"i. $OP1 Program All     " "  Programs all files (SA0,BL2,SA6,BL31 and u-boot)" \
 	"j. $OP4 eMMC boot setup " "  Configure an eMMC device for booting (only needed once)" \
 	"s. $FWR Show switches   " "  Show the switch settings for Renesas boards (in case you forgot)" \
+	"x1. $OP1 Erase BootPart1 " "  Erases eMMC Boot Partition 1 (needed when re-flashing to clear u-boot)" \
+	"x2. $OP1 Erase BootPart2 " "  Erases eMMC Boot Partition 2 (needed when re-flashing to clear u-boot env)" \
+	"x0. $OP1 Erase UserPart  " "  Erases eMMC User Partition 0 (optional)" \
```
Comment (commodo) [inline flash_writer_tool/flash_writer_tool.sh]: so, regarding point 5, this was already in the print_usage() function;

Comment (commodo) [reply flash_writer_tool/flash_writer_tool.sh]: oops;
 not here;

File: flash_writer_tool/flash_writer_tool.sh [hunk 2]
```diff
@@ -1360,6 +1387,9 @@ print_usage() {
 	echo "config file         # If no file specified, config.ini is assumed"
 	echo "$0 fw               # Downloads the flash writer program after RESET (must be run first)"
 	echo ""
+	echo "$0 x0               # erases User Partition Area on eMMC"
+	echo "$0 x1               # erases Boot Partition 1 on eMMC"
+	echo "$0 x2               # erases Boot Partition 2 on eMMC"
```
Comment (commodo) [inline flash_writer_tool/flash_writer_tool.sh:1392]: this is where they are listed in print_usage;

**Comment by commodo** [comment]
i've updated the PR;

unfortunately, i don't have a RZG2L board around for testing;
maybe i could get hold of one a bit later (in 1-2 weeks)

**Comment by seebe** [comment]
> i've updated the PR;

Merged.
I squashed all the changes into a single commit.
Thank you!

> unfortunately, i don't have a RZG2L board around for testing; maybe i could get hold of one a bit later (in 1-2 weeks)
I believe you that the erase is working on you board. As far as the script goes, it's doing what it is supposed to.
If you ever do get a G2L board, you can mention if it works/doesn't work. I will have a look some other time.

Thank you for your contribution.

**Comment by commodo** [comment]
> > i've updated the PR;
> 
> Merged. I squashed all the changes into a single commit. Thank you!
> 

oh right;
i forgot to mention that squashing is what i wanted;
i just went for small commits, to make the additions (from the previous review) easier to review in the next part;
then Github's squashing comes in handy;

> > unfortunately, i don't have a RZG2L board around for testing; maybe i could get hold of one a bit later (in 1-2 weeks)
> > I believe you that the erase is working on you board. As far as the script goes, it's doing what it is supposed to.
> > If you ever do get a G2L board, you can mention if it works/doesn't work. I will have a look some other time.
> 
> Thank you for your contribution.

Thank you :)

**Comment by seebe** [comment]
> i just went for small commits, to make the additions (from the previous review) easier to review in the next part;

I liked the small commits!  very easy and quick to review.

> then Github's squashing comes in handy;

I didn't even realize they had that option. It's nice.

**Pull request merged**